### ISSUE TITLE
next/link should be used with anchor as children

### ISSUE THREAD

**Issue by lfades** [vercel/commerce#330]
There are some places where `<Link>` is used without `<a>` as the children, this is bad for accessibility because links should be links. An example of this is the cart item, where the image and text go to the product page but without using an anchor, and right now the sidebar closes as soon as you click, instead of waiting for the navigation.

**Comment by vczb** [comment]
I've identified that the problem with the sidebar closing before the end of loading is actually because the `closeSidebarIfPresent()` method is executed on element click.


components/cart/CartItem/CartItem.tsx
